engine/collection/message_hub/message_hub.go

```diff
@@ -333,8 +348,12 @@ func (h *MessageHub) processQueuedProposal(header *flow.Header) error {
 
 	log.Debug().Msg("processing cluster broadcast request from hotstuff")
 
+	hotstuffProposal := model.ProposalFromFlow(header)
+	// notify vote aggregator that new block proposal is available, in case we are next leader
+	h.voteAggregator.AddBlock(hotstuffProposal)
+
 	// TODO(active-pacemaker): replace with pub/sub?
-	h.hotstuff.SubmitProposal(model.ProposalFromFlow(header)) // non-blocking
+	h.hotstuff.SubmitProposal(hotstuffProposal) // non-blocking
```

Comment:
I am not sure, but I am concerned about skipping the compliance engine here. One of the important tasks of the compliance engine is to persist blocks in the DB. 
I think it would help reduce complexity to treat our own blocks exactly the same as external blocks as much as we can. Thereby, we avoid extra code and special cases. In particular, I think it is totally fine for our own blocks to undergo verification. Other nodes have to do the same work before voting. I feel there is little use for the proposer to skip the verification, because the proposer is the only node that can do this, i.e. we don't speed up the consensus hot-path. I just don't think it is worth the extra complexity in the code.

Reply:
block will be added into DB by block builder but now I think it's a good idea to pass it through compliance engine and have a symmetrical approach for handling blocks, even though it does redundant job in terms of verification.

Reply:
While implementing this I've noticed that compliance engine [skips]() already processed proposals, maybe we would like to move block extension from builder to compliance engine exclusively

Reply:
Let's try to handle this as part of [this](https://github.com/onflow/flow-go/pull/3512) PR. It has strongly typed interface and it will be easier to replace it when experimenting with builder implementation.

Reply:
I added an issue https://github.com/onflow/flow-go/issues/3552 for later